src/drive/lib/encryption.js

```diff
@@ -121,7 +110,42 @@ export const importKey = async (
   )
 }
 
-export const generateAESKey = async ({ algorithm, keyLength } = {}) => {
+/**
+ * Wrap the given key with the wrappingKey in a JWE-like format.
+ * @param {CryptoKey} key  The key to wrap
+ * @param {CryptoKey} wrappingKey      The wrapping key used to encrypt the key. It must have the 'wrapKey' property
+ * @param {string} wrappingKeyId      The id of the wrapping key, used to retrieve it for unwrapping
+ * @param {object} params           Additional parameters
+ * @param {string} iv        The initialization vector of the key
+
+ * https://tools.ietf.org/html/rfc7516
+ */
+export const wrapAESKey = async (
+  key,
+  wrappingKey,
+  wrappingKeyId,
+  { iv } = {}
+) => {
+  const keyJwk = await window.crypto.subtle.exportKey('jwk', key)
+  const wrappingKeyJwk = await window.crypto.subtle.exportKey('jwk', key)
+  const encryptedKey = encode(
+    await window.crypto.subtle.wrapKey('raw', key, wrappingKey, {
+      name: 'AES-KW'
+    })
+  )
+  // "alg"" is the algorithm used to encrypt the key
+  // "enc" is the algorithm of the encrypted key
+  // "kid" is the id of the wrapping key
+  const header = {
+    alg: wrappingKeyJwk.alg,
+    enc: keyJwk.alg,
+    kid: wrappingKeyId
+  }
+  return { header, encrypted_key: encryptedKey, iv: iv }
```

Comment:
What about the `iv` encoding ? I do not master the data-format enough to see if it's OK to store the `iv` like that or if it's better to encode it first (like we did until now)

Reply:
IMHO, the iv must be encoded before being passed to the function

Reply:
OK